### ISSUE TITLE
Batch size and iteration

### ISSUE THREAD

**Issue by stemanik** [Deep-Learning-Person-Re-Identification/Implementaion-1#18]
Hi,
First of all thanks a lot for the code.
This is sort of an absolute beginner question, but if you use 150 batch size samples, and the total images of the CUHK03 dataset are 13164, the numbers of iteration per epoch should be 13164/batch size, right?. If you use 30000 iteration don't you risk to fall in the overfitting problem?
In the article, there was written that they use 210000 iteration, but they don't clarify if they use 210000 iterations per epoch or this number include all training iterations.

Thanks

**Comment by Ning-Ding** [comment]
Hi, to estimate how many iterations the model should be trained for this task is different with what you think.
First, though the total images of the CUHK03 dataset are about 13000, they are not the direct input of the model. The input of the model is a pair of two images, and has two labels: positive and negative. If one person has 10 images and each camera has 5 images, the total positive instances are 25. If the ratio of positive instance to negative instance is 1:3, the total instances for one person are 100. So it may have 130000 training examples. The total params of the model are nearly 20 millions so it's fine with about 100k iterations.
Hope this can help you.

**Comment by stemanik** [comment]
Ok, i understand the first part of your answer, but i still can not understand why with 20 millions params you need 100k iterations. what kind of formula you use to determine the iterations number?
Thank you so much for your help!

**Comment by Ning-Ding** [comment]
There may be a formula to compute the number of iterations but I don't know what it is. In my opinion, at least let the model see every instance 5-20 times. Besides, saving the interval model weights and watching the loss curve also give you some help.

**Comment by stemanik** [comment]
Thanks!!

**Comment by LG17** [comment]
Hi,
Thanks a lot for the code.
I wonder that if it's better to set the batch_size 128, or 32, 64 ? 
And how long does it take to run all the 100 * 30,000 epochs?

Thanks!

**Comment by PrernaPrem** [comment]
Hi , 
Can you please tell how many training samples you have used in the code and how many testing samples?

**Comment by hs-ucsb** [comment]
@Ning-Ding 
Hi,
Can you please explain what the parameters of train method (inside main.py) denotes?
